pymc/logprob/transforms.py

```diff
@@ -722,7 +738,9 @@ def backward(self, value, *inputs):
     def log_jac_det(self, value, *inputs):
         inv_power = 1 / self.power
         # Note: This fails for value==0
-        return np.log(np.abs(inv_power)) + (inv_power - 1) * at.log(value)
+        return at.switch(
```

Comment:
Have to think about this, odd powers should work with negative numbers, no?

The previous implementation returned -inf as well I think

Reply:
I see so would adding another check for the power work?
````python
return at.switch(
            (value >= 0) or (self.power % 2 != 0), np.log(np.abs(inv_power)) + (inv_power - 1) * at.log(value), -np.inf
        )
````

Reply:
We don't need a switch for the power. We only support constant powers, so you can use a simple python if else, like the backward does.

Reply:
But the problem is just the jacobian, it should probably take the log of abs of the value, but I didn't check it carefully.

Reply:
````python
return at.switch(
    at.or_(value >= 0,  self.power % 2 != 0), np.log(np.abs(inv_power)) + (inv_power - 1) * at.log(at.abs(value)), -np.inf
                 )
````
I'm running into problems with the above and its throwing:
`NotImplementedError: Logprob method not implemented for Elemwise{pow,no_inplace}`
I think its because `value >= 0` evaluates to `Elemwise{ge,no_inplace}.0` while  `self.power % 2 != 0` evaluates to `True`. 

Not sure on what the fix is here... any suggestions?

Reply:
Figured it out!

Reply:
Had to change the `backward` function in `PowerTransform` to allow for negative values, but everything seems to work as intended now.